pkg/vsphere/disk/disk_manager.go

```diff
@@ -108,10 +107,9 @@ func (m *Manager) CreateAndAttach(op trace.Operation, newDiskURI,
 		return nil, errors.Trace(err)
 	}
 
-	d.setAttached(devicePath)
-
-	if err := waitForPath(op, devicePath); err != nil {
-		op.Infof("waitForPath failed for %s with %s", newDiskURI, errors.ErrorStack(err))
+	blockDev, err := waitForDevice(op, devicePath)
```

Comment:
It is still possible that this will be preempted by the switch-root that breaks the tether. I do not think we have removed the infinite loop there which spins waiting for devices to present. This could also help us determine if the vm is actually stuck in that state right?